### DISCUSSION TITLE
Although documented in reality programmatically defined entity mappings are not supported

### DISCUSSION THREAD

**Discussion by frank-montyne** [infinispan/infinispan#11077] [Q&A]
In the IndexingConfigurationBuilder I found the following comment: "//TODO  [anistor] This does not take into account eventual programmatically defined entity mappings"

This effectively makes it impossible to define entity mappings programmatically. Am I the only one who wants to use this functionality? In my case I'm indexing objects that are not under my control so I cannot annotate them. Even if I would be able to annotate them we do not want to introduce dependencies on infinispan because they are used in other contexts also. I would assume other people have the same problems. 

Is there any chance this issue will get resolved soon?

Thanks

Below is the code in question.

 ```
  @Override
   public void validate() {
      if (enabled()) {
         //Indexing is not conceptually compatible with Invalidation mode
         if (clustering().cacheMode().isInvalidation()) {
            throw CONFIG.invalidConfigurationIndexingWithInvalidation();
         }
         if (indexedEntities().isEmpty() && !getBuilder().template()) {
            //TODO  [anistor] This does not take into account eventual programmatically defined entity mappings
            throw CONFIG.noIndexableClassesDefined();
         }
      } else {
         if (!indexedEntities().isEmpty()) {
            throw CONFIG.indexableClassesDefined();
         }
      }

```

**Comment by fax4ever** [comment]
Hi @frank-montyne yes probably the only use case I can see for programmatic mapping. is the one in which you cannot touch your entities with extra annotations (and dependencies).
Programmatic mapping on the other hand tends to be a little more verbose and the fields/properties need to be validated...
You can open an issue to support it.
But if you use ProtoStream probably you have the same issue with the ProtoStream annotations.

**Comment by frank-montyne** [comment]
Hi Fabio,

Thanks for the reply. I'll have a look to open a ticket for this. Any idea
if this is a big effort?

Best regards,
Frank

**Comment by fax4ever** [comment]
My major concern is about how to validate it...
Do you need it for the embedded case?

**Comment by frank-montyne** [comment]
Yes it's for the embedded case

**Reply by fax4ever** [reply]
For the remote case there is a special builder to create a custom builder using `org.infinispan.api.protostream.builder.ProtoBuf`. @frank-montyne please open an issue to support it with for the embedded case.
